src/Rating.js

```diff
@@ -1,8 +1,228 @@
+        this.delegate('click', 'a', this.handleClickEvent.bind(this));
+        this.on('mouseout', this.handleMouseOutEvent.bind(this));
+    }
+
+    /**
+     * Handles click event
+     * @param {Event} event
+     * @protected
+     */
+    handleClickEvent(event) {
+        if (!this.disabled) {
+            let index = parseInt(event.target.dataset.index, 10);
+
+            if (this.selectedIndex === index && this.canReset) {
+                this.reset();
+            }
+            else {
+                this.selectedIndex = index;
+            }
+        }
+    }
+
+    /**
+     * Reset rating attributes to its initial value
+     * @protected
+     */
+    reset() {
+        this.rate = null;
+        this.selectedIndex = -1;
+    }
+
+    /**
+     * Handles mouseout event
+     * @protected
+     */
+    handleMouseOutEvent() {
+        this.currentMouseTarget = undefined;
+        this.highlightStarIndex = this.selectedIndex;
+    }
+
+    /**
+     * Handles mouseover event
+     * @param {event} event
+     * @protected
+     */
+    handleMouseOverEvent(event) {
+        if (!this.disabled) {
+            let index = Number.parseInt(event.target.dataset.index, 10);
+
+            if (this.currentMouseTarget !== index) {
+                this.highlightStarIndex = index;
+            }
+
+            this.currentMouseTarget = index;
+        }
+    }
+
+    /**
+     * Syncronize seletected item attribute chagens
+     * @param {object} item
+     * @protected
+     */
+    syncSelectedItem(item) {
+        if (item) {
+            this.rate = item.value;
+        }
+    }
+
+    /**
+     * Current item selected index
+     * @param {number} index
+     * @protected
+     */
+    syncSelectedIndex(index) {
+        this.highlightStarIndex = index;
+        this.selectedItem = this.options[index] || null;
+    }
 }
 
+Rating.ATTRS = {
+
+    /**
+     * Block or unblock rating functionality
+     * @type {?boolean}
+     * @default false
+     */
+    cssClasses: {
+        value: {
+            off: 'glyphicon glyphicon-star-empty',
+            on: 'glyphicon glyphicon-star'
+        }
+    },
+
+    /**
+    * Block or unblock rating functionality
+    * @type {?boolean}
+    * @default false
+    */
+    disabled: {
+        value: false,
+        validator: core.isBoolean
+    },
+
+    /**
+    * The current rate
+    * @type {?number}
+    * @default 0
+    */
+    rate: {
+        value: null
+    },
+
+    /**
+    * The current highlighted index
+    * @type {number}
+    * @default -1
+    */
+    selectedIndex: {
+        validator: core.isNumber,
+        value: -1
+    },
+
+
+    /**
+    * The current highlighted index
+    * @type {number}
+    * @default -1
+    */
+    highlightStarIndex: {
+        value: -1
+    },
+
+    /**
+    * The current seleted item object
+    * @type {object}
+    * @default null
+    */
+    selectedItem: {
```

Comment:
Yes... I changed the logic. I removed these both attributes and I renamed the attribute rate to value. Value attribute now receive the index of the given options from 0 to options.length - 1. Its default value is -1 to indicate that any option is selected. 

Now you can set like this:

ratingInstance.value = 0 (the first star get light);
ratingInstance.value = 1 (the second star and the previous star get light, and so on)

We use value attribute to highlight the stars on mouseover event as well.

So... The highlight logic is all about the value attribute.

I think it's clearer now.

If the developer wants to get the given options.value value he can use rating.value to interact over options attribute: rating.options[rating.value];

What do you think about it?